airflow/models/taskinstance.py

```diff
@@ -927,6 +929,10 @@ def signal_handler(signum, frame):
                 start_time = time.time()
 
                 self.render_templates(context=context)
+                if STORE_SERIALIZED_DAGS:
+                    RTIF.write(RTIF(ti=self, render_templates=False))
```

Comment:
This function is odd/not sure I see the point of it:

```suggestion
                    session.merge(RTIF(ti=self, render_templates=False))
```

Reply:
Or if you want it to be a function make it `RTIF().write()` -- but we should re-use the session either way.

Reply:
Updated